File: app/backend/app/schemas/payment.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class PaymentRequestCreate(BaseModel):
    payment_type: str = "full"  # "full" | "deposit" | "balance"
    # Override amount — defaults to lead's quoted_price_total if omitted
    amount_override_cents: Optional[int] = None
    # Override phone — defaults to lead's customer_phone if omitted
    phone_override: Optional[str] = None


class PaymentOut(BaseModel):
    id: str
    lead_id: str
    amount_cents: int
    payment_type: str
    status: str
    payment_link_url: Optional[str]
    square_order_id: Optional[str]
    square_payment_id: Optional[str]
    sent_to_phone: Optional[str]
    sent_at: Optional[datetime]
    paid_at: Optional[datetime]
    created_by: Optional[str]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: app/frontend/src/types/payment.ts
export type PaymentStatus = 'pending' | 'paid' | 'failed' | 'refunded' | 'cancelled'
export type PaymentType = 'full' | 'deposit' | 'balance'

export interface LeadPayment {
  id: string
  lead_id: string
  amount_cents: number
  payment_type: PaymentType
  status: PaymentStatus
  payment_link_url: string | null
  square_order_id: string | null
  square_payment_id: string | null
  sent_to_phone: string | null
  sent_at: string | null
  paid_at: string | null
  created_by: string | null
  created_at: string
}

export interface PaymentRequestCreate {
  payment_type?: PaymentType
  amount_override_cents?: number
  phone_override?: string
}
